OnboardProject/Pages/Login/LoginPage.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import {
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+} from 'react-native';
+import { client, loginMutation } from '../../Utils/graphQlComm';
+import { storeData } from '../../Utils/LocalStorage';
+import { Navigation } from 'react-native-navigation';
+import { styles } from './Loginstyles';
+
+
+interface LoginState {
+  email: string,
+  password: string,
+  errorMessage: string,
+  hasEmailError: boolean,
+  hasPasswordError: boolean,
+  hasLoginError: boolean,
+  isLoading: boolean
+}
+interface LoginProps {
+  componentId: string
+}
+class LoginPage extends React.Component<LoginProps, LoginState>{
+
+  constructor(props: LoginProps) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      errorMessage: '',
+      hasEmailError: false,
+      hasPasswordError: false,
+      hasLoginError: false,
+      isLoading: false
+    }
+  }
+
+  handleLogin = () => {
+
+    const emailOk: Boolean = this.checkEmailFormat();
+    const passwordOk: Boolean = this.checkPasswordFormat();
+
+    if (emailOk && passwordOk) {
+
+      this.setState({ isLoading: true });
+
+      //Obtém objeto gqltag a partir do que foi preenchido no formulário
+      const login = loginMutation(this.state.email, this.state.password);
+
+      //Envia request
+      client.mutate({ mutation: login }).then((result) => {
+
+        storeData("AUTH_TOKEN", result.data.login.token);
+
+        this.setState({ isLoading: false });
+        this.setState({ hasEmailError: false, hasLoginError: false, hasPasswordError: false, errorMessage: "" })
+
+        Navigation.push(this.props.componentId, {
+          component: {
+            name: 'Users',
+            options: {
+              topBar: {
+                title: {
+                  text: 'Usuários'
+                }
+              }
+            }
+          }
+        })
+
+      }).catch((error) => {
+        this.setState({ isLoading: false });
+        console.log(error.graphQLError[0])
+        // let errorMessage = error.toString().split(':')[2].trim();
+
+        // this.setState({ hasLoginError: true, errorMessage: errorMessage });
+      });
```

Comment:
Promise também tem o `finally`
```suggestion
      client.mutate({ mutation: login }).then((result) => {

        storeData("AUTH_TOKEN", result.data.login.token);

        this.setState({ isLoading: false });
        this.setState({ hasEmailError: false, hasLoginError: false, hasPasswordError: false, errorMessage: "" })

        Navigation.push(this.props.componentId, {
          component: {
            name: 'Users',
            options: {
              topBar: {
                title: {
                  text: 'Usuários'
                }
              }
            }
          }
        })

      }).catch((error) => {
        this.setState({ isLoading: false });
        console.log(error.graphQLError[0])
        // let errorMessage = error.toString().split(':')[2].trim();

        // this.setState({ hasLoginError: true, errorMessage: errorMessage });
      })
      . finally(() => {
        this.setState({ isLoading: false });
      });
```